### PR TITLE
P2-704 Same Post and Job titles do not trigger red analysis

### PR DESCRIPTION
**Pull request by hansjovis**
## Summary

This PR can be summarized in the following changelog entry:

* [schema-blocks] Reworks the imports and exports of the schema blocks package.

## Relevant technical choices:

* Decided together with @increddibelly to rework the imports and exports.
* Reworked the warning messages in the sidebar.
	* Now gets the warnings from the `BlockValidationResult`s itself, rather than a separate result -> warning map.

## Test instructions

This PR can be tested by following these steps:

* Companion PR of https://github.com/Yoast/wordpress-seo-premium/pull/3261, please follow its test instructions.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
	* See PR in Premium: https://github.com/Yoast/wordpress-seo-premium/pull/3261

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #

### CODE REVIEW

File: packages/schema-blocks/src/functions/presenters/InnerBlocksSidebarPresenter.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { ReactElement } from "react";
 import { createElement } from "@wordpress/element";
 import { BlockInstance } from "@wordpress/blocks";
-import BlockSuggestions from "../../blocks/BlockSuggestions";
+import { RequiredBlocks as BlockSuggestions } from "../../blocks/BlockSuggestions";
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/InnerBlocksSidebarPresenter.tsx:4]: why the rename?

File: packages/schema-blocks/src/functions/validators/index.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import attributeExists from "./attributeExists";
 import attributeNotEmpty from "./attributeNotEmpty";
-import getInvalidInnerBlocks from "./innerBlocksValid";
+import { validateInnerBlocks as getInvalidInnerBlocks } from "./innerBlocksValid";
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/validators/index.ts:3]: why the rename?
```suggestion
import { validateInnerBlocks } from "./innerBlocksValid";
```

File: packages/schema-blocks/src/functions/validators/innerBlocksValid.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import recurseOverBlocks from "../blocks/recurseOverBlocks";
 import { getInnerblocksByName } from "../innerBlocksHelper";
 import logger from "../logger";
 import isValidResult from "./isValidResult";
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/validators/innerBlocksValid.ts:8]: there's still a default export remaining here.

File: packages/schema-blocks/src/instructions/blocks/SidebarSelect.ts [hunk 4]
```diff
@@ -3,10 +3,10 @@ import { BlockEditProps, BlockConfiguration } from "@wordpress/blocks";
 import { createElement } from "@wordpress/element";
 import { SelectControl } from "@wordpress/components";
 // Internal imports.
```
Comment (increddibelly) [inline packages/schema-blocks/src/instructions/blocks/SidebarSelect.ts:5]:
```suggestion
```

File: packages/schema-blocks/src/instructions/blocks/index.ts [hunk 5]
```diff
@@ -14,3 +14,7 @@ import "./TextInput";
 import "./VariableTagRichText";
 import "./Variation";
 import "./VariationPicker";
+
+import VariableTagRichText from "./VariableTagRichText";
+
+export { VariableTagRichText };
```
Comment (increddibelly) [inline packages/schema-blocks/src/instructions/blocks/index.ts:20]: why import the rest if we're not exporting them?

File: packages/schema-blocks/src/instructions/schema/index.ts [hunk 6]
```diff
@@ -7,4 +7,7 @@ import "./InnerBlocksID";
 import "./JobEmploymentType";
 import "./List";
 import "./Permalink";
-import "./Schema";
+import SchemaInstruction from "./Schema";
+
+// Need to export something for the side-effects of the imports to work...
```
Comment (increddibelly) [inline packages/schema-blocks/src/instructions/schema/index.ts:12]: dank voor dit comment! 
kunnen we de side effects expliciet maken / aanroepen?

File: packages/schema-blocks/tests/functions/BlocksHelper.test.ts [hunk 7]
```diff
@@ -1,13 +1,21 @@
 import { dispatch, select } from "@wordpress/data";
 import { BlockInstance } from "@wordpress/blocks";
 
-import { removeBlock, restoreBlock, getBlockType } from "../../src/functions/BlockHelper";
+import { removeBlock, restoreBlock, getBlockType, getHumanReadableBlockName } from "../../src/functions/BlockHelper";
```
Comment (increddibelly) [inline packages/schema-blocks/tests/functions/BlocksHelper.test.ts:4]: rename the file from BlocksHelper.test.ts to BlockHelper.test.ts (singular)

File: packages/schema-blocks/tests/functions/presenters/SidebarWarningPresenter.test.ts [hunk 8]
```diff
@@ -1,8 +1,7 @@
 import { BlockValidation, BlockValidationResult } from "../../../src/core/validation";
-import getWarnings, { createAnalysisMessages, sanitizeBlockName } from "../../../src/functions/presenters/SidebarWarningPresenter";
+import getWarnings, { createAnalysisMessages } from "../../../src/functions/presenters/SidebarWarningPresenter";
```
Comment (increddibelly) [inline packages/schema-blocks/tests/functions/presenters/SidebarWarningPresenter.test.ts:2]: can we un-default export the getWarnings?

**Review by increddibelly — CHANGES_REQUESTED** [review]
only a few comments for minor improvements.

also, /src/functions/blocks/index.ts is an empty file, please remove

File: packages/schema-blocks/src/instructions/blocks/index.ts (hunk 5 above)
Comment (hansjovis) [reply packages/schema-blocks/src/instructions/blocks/index.ts:20]: For the `BlockInstruction.register` side effects.

Without them the block instructions are not registered.

Comment (hansjovis) [reply packages/schema-blocks/src/instructions/blocks/index.ts:20]: But yes, I will export them all.

**Comment by hansjovis** [comment]
Closed in favour of https://github.com/Yoast/javascript/pull/1129.

**Pull request closed without merging**